apps/reaction/tests/integration/api/mutations/checkout/promotionCheckout.test.js

```diff
@@ -0,0 +1,268 @@
+
+  await testApp.collections.Promotions.insertOne(mockPromotion);
+});
+
+// There is no need to delete any test data from collections because
+// testApp.stop() will drop the entire test database. Each integration
+// test file gets its own test database.
+afterAll(() => testApp.stop());
+
+describe("Promotions", () => {
+  let cartToken;
+  let opaqueCartId;
+  let opaqueCartProductVariantId;
+  let opaqueFulfillmentGroupId;
+  let opaqueFulfillmentMethodId;
+  let latestCartSummary;
+
+  beforeAll(async () => {
+    opaqueCartProductVariantId = encodeProductOpaqueId(internalVariantIds[1]);
+    await testApp.clearLoggedInUser();
+  });
+
+  const shippingAddress = {
+    address1: "12345 Drive Lane",
+    city: "The city",
+    country: "USA",
+    firstName: "FName",
+    fullName: "FName LName",
+    isBillingDefault: false,
+    isCommercial: false,
+    isShippingDefault: false,
+    lastName: "LName",
+    phone: "5555555555",
+    postal: "97878",
+    region: "CA"
+  };
+
+  test("create a new cart", async () => {
+    const result = await createCart({
+      createCartInput: {
+        shopId: opaqueShopId,
+        items: {
+          price: {
+            amount: 19.99,
+            currencyCode: "USD"
+          },
+          productConfiguration: {
+            productId: opaqueProductId,
+            productVariantId: opaqueCartProductVariantId
+          },
+          quantity: 6
+        }
+      }
+    });
+
+    cartToken = result.createCart.token;
+    opaqueCartId = result.createCart.cart._id;
+  });
+
+  test("set email on anonymous cart", async () => {
+    const result = await setEmailOnAnonymousCart({
+      input: {
+        cartId: opaqueCartId,
+        cartToken,
+        email: "test@email.com"
+      }
+    });
+
+    opaqueCartId = result.setEmailOnAnonymousCart.cart._id;
+  });
+
+  test("set shipping address on cart", async () => {
+    const result = await setShippingAddressOnCart({
+      input: {
+        cartId: opaqueCartId,
+        cartToken,
+        address: {
+          address1: "12345 Drive Lane",
+          city: "The city",
+          country: "USA",
+          firstName: "FName",
+          fullName: "FName LName",
+          lastName: "LName",
+          phone: "5555555555",
+          postal: "97878",
+          region: "CA"
+        }
+      }
+    });
+
+    opaqueFulfillmentGroupId = result.setShippingAddressOnCart.cart.checkout.fulfillmentGroups[0]._id;
+  });
+
+  test("get available fulfillment options", async () => {
+    const result = await updateFulfillmentOptionsForGroup({
+      input: {
+        cartId: opaqueCartId,
+        cartToken,
+        fulfillmentGroupId: opaqueFulfillmentGroupId
+      }
+    });
+
+    const option = result.updateFulfillmentOptionsForGroup.cart.checkout.fulfillmentGroups[0].availableFulfillmentOptions[0];
+    opaqueFulfillmentMethodId = option.fulfillmentMethod._id;
+  });
+
+  test("select the `Standard mockMethod` fulfillment option", async () => {
```

Comment:
These are set up like separate tests but in fact there is only one test, which should describe what it's testing, e.g. "place an order with discount and get the correct values"